app/controllers/HistoricalExchangeRatesController.rb

```diff
@@ -0,0 +1,22 @@
+class HistoricalExchangeRatesController < ApplicationController
+    def index
+      rates = HistoricalExchangeRate.all
+  
+      # Filtrar por rango de fechas
+      if params[:start_date].present? && params[:end_date].present?
+        rates = rates.where(date: params[:start_date]..params[:end_date])
+      end
```

Comment:
Me llama la atención que sólo permite filtrar por fechas cuando se indican ambas: inicio y fin. Sería fácil permitir filtrar pasando sólo fecha de inicio, sólo fin, o ambas:
```suggestion
      rates = rates.where(date: params[:start_date]..) if params[:start_date]
      rates = rates.where(date: ..params[:end_date]) if params[:end_date]
```
Y lo mismo con el `from_currency` y `to_currency`.

Es correcto si la especificación pedía exactamente esto. Si no, considero preferible mi propuesta, que es más flexible sin añadir complejidad